content/links-api.md

```diff
@@ -0,0 +1,225 @@
+
+!!! note
+    The links API is available in bosh-release v266.0+
+
+    This API reference assumes the director is deployed with UAA.
+
+See [Links](links.md) for an introduction on links, link providers, and link consumers. This API is hosted on the director.
+
+## Providers
+### `GET /link_providers`: List Providers
+
+Obtain an array of providers created in a deployment.
+
+#### Request Query Params
+* **deployment**: [String] Deployment name.
+
+```bash
+$ uaac curl https://10.0.1.6:25555/link_providers?deployment=zookeeper
+[
+  {
+    "owner_object": {
+      "info": {
+        "instance_group": "zookeeper"
+      },
+      "name": "zookeeper",
+      "type": "job"
+    },
+    "link_provider_definition": {
+      "name": "peers",
+      "type": "zookeeper_peers"
+    },
+    "deployment": "zookeeper",
+    "shared": false,
+    "name": "peers",
+    "id": "2"
+  },
+  {
+    "owner_object": {
+      "info": {
+        "instance_group": "zookeeper"
+      },
+      "name": "zookeeper",
+      "type": "job"
+    },
+    "link_provider_definition": {
+      "name": "conn",
+      "type": "zookeeper"
+    },
+    "deployment": "zookeeper",
+    "shared": true,
+    "name": "conn",
+    "id": "1"
+  }
+]
+```
+
+## Consumers
+
+### `GET /link_consumers`: List Consumers
+
+Obtain an array of consumers created for a deployment.
+
+#### Request Query Params
+* **deployment**: [String] Deployment name.
+
+```bash
+$ uaac curl https://10.0.1.6:25555/link_consumers?deployment=zookeeper
+[
+  {
+    "link_consumer_definition": {
+      "type": "zookeeper_peers",
+      "name": "peers"
+    },
+    "owner_object": {
+      "info": {
+        "instance_group": "zookeeper"
+      },
+      "name": "zookeeper",
+      "type": "job"
+    },
+    "deployment": "zookeeper",
+    "optional": false,
+    "name": "peers",
+    "id": "1"
+  },
+  {
+    "link_consumer_definition": {
+      "type": "zookeeper",
+      "name": "conn"
+    },
+    "owner_object": {
+      "info": {
+        "instance_group": "smoke-tests"
+      },
+      "name": "smoke-tests",
+      "type": "job"
+    },
+    "deployment": "zookeeper",
+    "optional": false,
+    "name": "conn",
+    "id": "2"
+  }
+]
+
+```
+
+## Links
+
+### `GET /links`: List Links
+
+Obtain an array of links created for a deployment.
+
+#### Request Query Params
+* **deployment**: [String] Deployment name.
+
+```bash
+$ uaac curl https://10.0.1.6:25555/links?deployment=zookeeper
+[
+  {
+    "created_at": "2018-08-14 18:00:36 UTC",
+    "link_provider_id": "4",
+    "link_consumer_id": "3",
+    "name": "peers",
+    "id": "1"
+  },
+  {
+    "created_at": "2018-08-14 18:00:36 UTC",
+    "link_provider_id": "3",
+    "link_consumer_id": "4",
+    "name": "conn",
+    "id": "2"
+  }
+]
+```
+
+### `POST /links`: Create Link
+
+Create an external link with a user-defined consumer. Uses an existing provider.
+
+!!! note
+    The UAA client creating the link must have the `create_link` permission.
```

Comment:
Looking at the code, `create_link` and `delete_link` are only an internal representation used by the director code. From a UAA client perspective, it looks like the client should have either an bosh admin scope or a bosh team admin authorities.

https://github.com/cloudfoundry/bosh/blob/6f5263dcaf0f4f09923b96cadf60a965b843a7da/src/bosh-director/lib/bosh/director/permission_authorizer.rb#L79-L80